Applications/Spire/Include/Spire/Ui/QuantityBox.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef SPIRE_QUANTITY_BOX_HPP
+#define SPIRE_QUANTITY_BOX_HPP
+#include "Nexus/Definitions/Quantity.hpp"
+#include "Spire/Ui/DecimalBoxAdaptor.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /** A ScalarValueModel over Quantity values. */
+  using QuantityModel = ScalarValueModel<Nexus::Quantity>;
+
+  /** A ScalarValueModel over optional Quantity values. */
+  using OptionalQuantityModel =
+    ScalarValueModel<boost::optional<Nexus::Quantity>>;
+
+  /** A LocalScalarValueModel over Quantity values. */
+  using LocalQuantityModel = LocalScalarValueModel<Nexus::Quantity>;
+
+  /** A LocalScalarValueModel over optional Quantity values. */
+  using LocalOptionalQuantityModel =
+    LocalScalarValueModel<boost::optional<Nexus::Quantity>>;
+
+  class QuantityBox : public DecimalBoxAdaptor<Nexus::Quantity> {
+    public:
+
+      /**
+       * Constructs a QuantityBox with a LocalValueModel.
+       * @param modifiers The keyboard modifier increments.
+       * @param parent The parent widget.
+       */
+      explicit QuantityBox(QHash<Qt::KeyboardModifier,
```

Comment:
Break lines up at the higher nesting level.

Reply:
Can you explain what you mean? I can't find an issue with this formatting.

Reply:
explicit QuantityBox(
      QHash<Qt::KeyboardModifier, Nexus::Quantity> modifiers, QWidget* parent = nullptr);